homeassistant/components/usb/__init__.py

```diff
@@ -410,6 +459,19 @@ async def _async_process_ports(self, ports: Sequence[ListPortInfo]) -> None:
                 )
             ]
 
+        unique_devices = set(usb_devices)
```

Comment:
If there are duplicates, it probably doesn't make sense to call back the same device twice below so it would probably make sense to make this a set from the get go